packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -92,6 +99,20 @@ function BlockPattern( {
 							onDragEnd( event );
 						}
 					} }
+					onClick={ () => {
+						setActivePattern( id ); // Set active pattern when clicked
+						onClick( pattern, blocks );
+						onHover?.( null );
+					} }
+					onKeyDown={ ( event ) => {
+						// Simulate button click with Enter or Space key press
+						if ( event.key === 'Enter' || event.key === ' ' ) {
+							event.preventDefault(); // Prevent default behavior for spacebar
+							setActivePattern( id );
+							onClick( pattern, blocks );
+							onHover?.( null );
+						}
+					} }
```

Comment:
As mentioned above, `Composite.Item` is already an interactive element, and all interaction-related code should happen there.

Reply:
Thanks for mentioning this point @ciampo 

I will update this line of codes by removing it since these are already handling in Composite component.

onKeyDown will add inside Composite.Item if required.

Reply:
I don't think we need the `onKeyDown` — see answer in https://github.com/WordPress/gutenberg/pull/65917#discussion_r1794960682

Reply:
Updated.